### ISSUE TITLE
Inconsistent order/timing of summon triggers (Steward of Darkshire & Sword of Justice)

### ISSUE THREAD

**Issue by rnilva** [HearthSim/hs-bugs#899]
![hearthstonedecktracker_2018-02-02_05-01-30](https://user-images.githubusercontent.com/28765421/35704094-9f9a134e-07e1-11e8-9cd9-473f690c8a8d.png)
![hearthstonedecktracker_2018-02-02_05-02-12](https://user-images.githubusercontent.com/28765421/35704095-9fc12d44-07e1-11e8-9079-fb3ebc49a072.png)

Regardless of the order of play, when a minion is summoned by playing a minion card, always "Whenever you summon..." cards' effect is triggered first so the Steward gives the Thalnos a Divine Shield buff successfully.

However, when minions are (a minions is) summoned by other effects (e.g. Hero Power, Deathrattle, Spell ...) always "After you summon..." cards' effect trigger each time when a minion is summoned, and after all the minion is summoned, the "Whenever... " cards' effect triggers.
so the Steward couldn't give the buff to spiders.

I saw #35  but it seems there is certain distinction between two kinds of summon triggers.
If an effect summons a bunch of minions (e.g. Muster for Battle, Call to Arms),
"Whenever..." guys always do their tasks after all the minions are summoned, contrast to Knife Juggler's behaviour in #861

**Comment by rnilva** [comment]
**Before reading
There are two kinds of Summon-related triggers : Pre-summon triggers and After-summon triggers.
The text of former ones described as "Whenever you summon a minion ..." and the latter ones have text of "After you summon a minion ...". 


I expected from the texts of cards that the **pre-summons should be handled being earlier** than the after summon cards, regardless of the order of play. In short,
What I expected : "Steward of Darkshire always gives a minion D.S. whether Sword of Justice is there or not"

In reality: 
- When I **PLAY** a minion with 1-health (Thalnos in SC) => works as expected.
- When minions are summoned by card effects => *after-summon triggers are always processed first*

**Comment by PattuX** [comment]
Basically #108

To clarify: Steward of Darkshire isn't a pre-summon trigger but an on-summon trigger. Those trigger in between Phases rather than having their own phase. To quote the advanced rulebook,
 leaving out some details (https://hearthstone.gamepedia.com/Advanced_rulebook#Playing.2Fsummoning_a_minion): 

When you Play a minion, the following Sequence takes place:

* The card is removed from your hand and its Mana cost is paid.
* On Play Phase: "Whenever you play..."
* (Because of the Summon Resolution Step,Steward of Darkshire queues and resolves for the played minion here.)
* Battlecry Phase
* After Play Phase: "After you play..."
* After Summon Phase: "After you summon..."

If a minion is merely summoned rather than played, it is inside of a larger Phase, so we are currently not processing Deaths or checking for win/loss/draw. The Sequence is now far simpler:

* The card is generated (or moved from your hand/deck if due to a Force Play mechanic) and enters Play, creating a Summon Event and becoming interactible.
* After Summon Phase
* (Because of the Summon Resolution Step,Steward of Darkshire queues and resolves for the played minion here.)

In other words, Steward of Darkshire always triggers after the first phase of the main sequence. For summoned minions this happens to be after the After Summon Phase, which I agree seems weird and should've been changed by now.

(Somewhat outdated) video on this topic: https://www.youtube.com/watch?v=QC8_jNOvdEs

The main principle explained in the video still works the same, however some cards (Coldarra Drake for example) no longer trigger in the SRS.

---

Short word on Pre-summon triggers: Those were only Cobalt Guardian and Murloc Tidecaller and they triggered before the minion even entered play, making them inconsistent with all other triggers that read "Whenever you summon". Also they had some weird bugs with Mirror Entity as you can read in the issue you linked.
